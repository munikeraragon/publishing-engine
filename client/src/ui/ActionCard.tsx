export interface ActionCardProps {
    src: string;
    title: string;
    description: string;
    className?: string;
    github?: string;
    live?: string;
}

export const ActionCard: React.FC<ActionCardProps> = ({
    src,
    title,
    description,
    className = '',
    github,
    live
}) => {
    return (
        <div className={`border border-gray-300 rounded-lg ${className}`}>
            <div className='flex bg-blue-100 rounded-lg'>
                <img className='mx-auto object-cover rounded-lg' src={src} />
            </div>

            <div className='m-3'>
                <div className='flex'>
                    <h2 className='text-lg mb-2'>{title}</h2>
                    <div className='flex-grow' />

                    {github && (
                        <a href={github} data-testid='github-link'>
                            <svg
                                xmlns='http://www.w3.org/2000/svg'
                                className='mx-1 h-5 w-5'
                                viewBox='0 0 24 24'>
                                <path d='M12 0c-6.626 0-12 5.373-12 12 0 5.302 3.438 9.8 8.207 11.387.599.111.793-.261.793-.577v-2.234c-3.338.726-4.033-1.416-4.033-1.416-.546-1.387-1.333-1.756-1.333-1.756-1.089-.745.083-.729.083-.729 1.205.084 1.839 1.237 1.839 1.237 1.07 1.834 2.807 1.304 3.492.997.107-.775.418-1.305.762-1.604-2.665-.305-5.467-1.334-5.467-5.931 0-1.311.469-2.381 1.236-3.221-.124-.303-.535-1.524.117-3.176 0 0 1.008-.322 3.301 1.23.957-.266 1.983-.399 3.003-.404 1.02.005 2.047.138 3.006.404 2.291-1.552 3.297-1.23 3.297-1.23.653 1.653.242 2.874.118 3.176.77.84 1.235 1.911 1.235 3.221 0 4.609-2.807 5.624-5.479 5.921.43.372.823 1.102.823 2.222v3.293c0 .319.192.694.801.576 4.765-1.589 8.199-6.086 8.199-11.386 0-6.627-5.373-12-12-12z' />
                            </svg>
                        </a>
                    )}
                    {live && (
                        <a href={live} data-testid='live-link'>
                            <svg
                                xmlns='http://www.w3.org/2000/svg'
                                className='mx-1 h-5 w-5'
                                viewBox='0 0 20 20'
                                fill='currentColor'>
                                <path d='M10 12a2 2 0 100-4 2 2 0 000 4z' />
                                <path
                                    fillRule='evenodd'
                                    d='M.458 10C1.732 5.943 5.522 3 10 3s8.268 2.943 9.542 7c-1.274 4.057-5.064 7-9.542 7S1.732 14.057.458 10zM14 10a4 4 0 11-8 0 4 4 0 018 0z'
                                    clipRule='evenodd'
                                />
                            </svg>
                        </a>
                    )}
                </div>

                <p className='font-light font-mono text-sm text-gray-700 hover:text-gray-900 transition-all duration-200'>
                    {description}
                </p>
            </div>
        </div>
    );
};
