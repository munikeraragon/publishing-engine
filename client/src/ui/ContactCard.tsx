import { TextInput } from './TextInput';
import { SelectInput } from './SelectInput';
import countries from '../constants/countries.json';

export interface ContactCardProps {
    firstName: string;
    lastName: string;
    email: string;
    country: string;
    phone: string;
    company: string;
    message: string;
    submitting: boolean;
    setFirstName: (value: string) => void;
    setLastName: (value: string) => void;
    setEmail: (value: string) => void;
    setCountry: (value: string) => void;
    setPhone: (value: string) => void;
    setCompany: (value: string) => void;
    setMessage: (value: string) => void;
    onSubmit: () => void;
}

export const ContactCard: React.FC<ContactCardProps> = ({
    firstName,
    lastName,
    email,
    country,
    phone,
    company,
    message,
    submitting,
    setFirstName,
    setLastName,
    setEmail,
    setCountry,
    setPhone,
    setCompany,
    setMessage,
    onSubmit
}) => {
    return (
        <div className='shadow-2xl grid grid-cols-1 lg:grid-cols-2 gap-4 p-8'>
            <div>
                <TextInput
                    label='First name'
                    name='First name'
                    placeHolder='Your first name'
                    value={firstName}
                    required={true}
                    className='my-2'
                    onChange={(event) => {
                        setFirstName(event.target.value);
                    }}
                />
            </div>

            <div>
                <TextInput
                    label='Last name'
                    name='Last name'
                    placeHolder='Your last name'
                    value={lastName}
                    required={true}
                    className='my-2'
                    onChange={(event) => {
                        setLastName(event.target.value);
                    }}
                />
            </div>

            <div>
                <TextInput
                    label='Email'
                    name='Email'
                    placeHolder='example@acme.com'
                    value={email}
                    required={true}
                    className='my-2'
                    onChange={(event) => {
                        setEmail(event.target.value);
                    }}
                />
            </div>

            <div>
                <SelectInput
                    label='Country'
                    name='Country'
                    value={country}
                    required={true}
                    options={countries}
                    className='my-2'
                    onChange={(event) => {
                        setCountry(event.target.value);
                    }}
                />
            </div>

            <div>
                <TextInput
                    label='Phone number'
                    name='Phone bumber'
                    placeHolder='(555) 555-5555'
                    value={phone}
                    required={true}
                    className='my-2'
                    onChange={(event) => {
                        setPhone(event.target.value);
                    }}
                />
            </div>

            <div>
                <TextInput
                    label='Company'
                    name='Company'
                    placeHolder='Acme Inc.'
                    value={company}
                    className='my-2'
                    onChange={(event) => {
                        setCompany(event.target.value);
                    }}
                />
            </div>

            <div className='col-span-2'>
                <TextInput
                    label='How can we help you?'
                    name='Message'
                    area={true}
                    value={message}
                    required={true}
                    className='my-2'
                    onChange={(event) => {
                        setMessage(event.target.value);
                    }}
                />
            </div>

            <div className='col-span-2'>
                <p className='text-sm mb-2 text-gray-600'>
                    By clicking &quot;Submit,&quot; I acknowledge receipt of the CodeGrow Privacy
                    Policy.
                </p>

                <p className='text-sm my-2 text-pink-600'>
                    Fields marked with an asterisk (*) are required.
                </p>
            </div>

            <div className='col-span w-48'>
                <button
                    onClick={onSubmit}
                    className={`${
                        submitting ? 'opacity-60' : ''
                    } w-full flex items-center justify-center px-8
                            py-2 border border-transparent text-base font-medium
                            rounded-md text-white bg-indigo-600 hover:bg-indigo-700
                            md:text-lg md:px-10`}>
                    {submitting ? 'Submiting...' : 'Submit'}
                </button>
            </div>
        </div>
    );
};
