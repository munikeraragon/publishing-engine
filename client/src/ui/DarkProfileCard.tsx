export interface ProfileCardProps {
    userIcon: string | undefined;
    userName: string | undefined;
    className?: string;
}

export interface EskeletonProps {
    className?: string;
}

export const ProfileCard: React.FC<ProfileCardProps> = ({ userIcon, userName, className = '' }) => {
    if (!userIcon || !userName) return <ProfileCardEskeleton className={className} />;
    return (
        <div className={`px-6 py-8 bg-gray-800 text-white rounded-lg ${className}`}>
            <div className='space-y-4 xl:space-y-6'>
                <img src={userIcon} alt='user profile' className='w-36 h-36 mx-auto rounded-full' />

                <div className='space-y-2'>
                    <div className='flex justify-center items-center flex-col space-y-3 text-lg font-medium leading-6'>
                        <h3>{userName}</h3>
                    </div>
                </div>
            </div>

            <div className='mb-2'>
                <h3 className='text-sm text-indigo-300 font-medium'>Work</h3>
                <p>Technical Director at Valtech </p>
            </div>

            <div className='mb-2'>
                <h3 className='text-sm text-indigo-300 font-medium'>Location</h3>
                <p>Copenhagen</p>
            </div>

            <div className='mb-2'>
                <h3 className='text-sm text-indigo-300 font-medium'>Joined</h3>
                <p>Nov 16, 2020</p>
            </div>
        </div>
    );
};

export const ProfileCardEskeleton: React.FC<EskeletonProps> = ({ className }) => {
    return (
        <div className={`px-6 py-10 text-center rounded-lg shadow-md ${className}`}>
            <div className='space-y-4 xl:space-y-6'>
                <div className='w-40 h-40 mx-auto rounded-full bg-gray-300 animate-pulse' />
                <div className='space-y-2'>
                    <div className='flex justify-center items-center flex-col space-y-3 text-lg font-medium leading-6'>
                        <span className='w-32 h-2 bg-gray-300 animate-pulse rounded-md my-1' />
                        <span className='w-20 h-2 bg-gray-300 animate-pulse rounded-md my-1' />
                    </div>
                </div>
            </div>
        </div>
    );
};
