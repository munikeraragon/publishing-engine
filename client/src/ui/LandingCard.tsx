import React from 'react';
import Link from 'next/link';

export interface LandingCardProps {
    className?: string;
    primaryHeader: string;
    secondaryHeader?: string;
    description?: string;
    linkLabel1?: string;
    linkLabel2?: string;
    linkHref1?: string;
    linkHref2?: string;
}

export const LandingCard: React.FC<LandingCardProps> = ({ className = '', ...props }) => {
    return (
        <div className={`flex flex-col justify-center mx-4 ${className}`}>
            <div className='sm:text-center lg:text-left'>
                <h1 className='text-4xl tracking-tight font-bold text-gray-700'>
                    {props.primaryHeader}
                </h1>

                {props.secondaryHeader && (
                    <h1 className='text-4xl font-bold text-indigo-600'>{props.secondaryHeader}</h1>
                )}

                {props.description && (
                    <p className='mt-3 text-base text-gray-500 sm:mt-5 sm:text-lg sm:max-w-xl sm:mx-auto md:mt-5 md:text-xl lg:mx-0'>
                        {props.description}
                    </p>
                )}
                <div className='mt-5 sm:mt-8 sm:flex sm:justify-center lg:justify-start'>
                    {props.linkLabel1 && (
                        <div className='z-10 rounded-md shadow'>
                            <Link href={props.linkHref1 || ''}>
                                <span className='w-full flex items-center justify-center px-8 py-2 border border-transparent text-base font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-700  md:py-2 md:text-lg md:px-10'>
                                    {props.linkLabel1}
                                </span>
                            </Link>
                        </div>
                    )}
                    {props.linkLabel2 && (
                        <div className='z-10 mt-3 sm:mt-0 sm:ml-3'>
                            <Link href={props.linkHref2 || 'link-2'}>
                                <span className=' w-full flex items-center justify-center px-8 py-3 border border-transparent text-base font-medium rounded-md text-indigo-700 bg-indigo-100 hover:bg-indigo-200 hover:bg-blue-200 md:py-2 md:text-lg md:px-10'>
                                {props.linkLabel2}
                                </span>
                            </Link>
                        </div>
                    )}
                </div>
            </div>
        </div>
    );
};
