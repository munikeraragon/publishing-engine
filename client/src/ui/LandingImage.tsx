
export interface LandingImageProps {
    className?: string
    src?: string
    alt?: string
}

export const LandingImage: React.FC<LandingImageProps> = ({ className="", ...props}) => {
  return (
      <div className={`flex flex-col justify-center ${className}`}>
          <img
              className="m-auto object-cover"
              src={props.src}
              alt={props.alt}
              style={{ maxHeight: "500px" }}
          />
      </div>
  );
};