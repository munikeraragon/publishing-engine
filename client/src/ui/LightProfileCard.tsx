export interface ProfileCardProps {
    userIcon: string | undefined;
    userName: string | undefined;
    className?: string;
}

export interface EskeletonProps {
    className?: string;
}

export const ProfileCard: React.FC<ProfileCardProps> = ({ userIcon, userName, className = '' }) => {
    if (!userIcon || !userName) return <ProfileCardEskeleton className={className} />;
    return (
        <div
            className={`px-6 py-4 text-gray-900 rounded-lg ${className}`}
            style={{
                border: 'solid 2px rgb(95, 95, 96)',
                boxShadow: 'rgb(95, 95, 96) 4px 4px 0px 0px',
                background: 'rgb(251, 251, 252)'
            }}>
            <div className='space-y-4 xl:space-y-6'>
                <img src={userIcon} alt='user profile' className='w-36 h-36 mx-auto rounded-full' />

                <div className='space-y-2'>
                    <div className='flex justify-center items-center flex-col space-y-3 text-lg leading-6'>
                        <h3>{userName}</h3>
                    </div>
                </div>
            </div>

            <div className='mb-2'>
                <h3 className='text-sm text-gray-500 font-medium'>Work</h3>
                <p>--------------------------</p>
            </div>

            <div className='mb-2'>
                <h3 className='text-sm text-gray-500 font-medium'>Location</h3>
                <p>------------</p>
            </div>

            <div className='mb-2'>
                <h3 className='text-sm text-gray-500 font-medium'>Joined</h3>
                <p>Nov 16, 2020</p>
            </div>
            <div className='text-center'>
                <button
                    className='bg-gray-800 hover:bg-gray-900 text-white
             font-medium rounded-md  py-1.5 px-8'>
                    Follow
                </button>
            </div>
        </div>
    );
};

export const ProfileCardEskeleton: React.FC<EskeletonProps> = ({ className }) => {
    return (
        <div className={`px-6 py-10 text-center rounded-lg shadow-md ${className}`}>
            <div className='space-y-4 xl:space-y-6'>
                <div className='w-40 h-40 mx-auto rounded-full bg-gray-300 animate-pulse' />
                <div className='space-y-2'>
                    <div className='flex justify-center items-center flex-col space-y-3 text-lg font-medium leading-6'>
                        <span className='w-32 h-2 bg-gray-300 animate-pulse rounded-md my-1' />
                        <span className='w-20 h-2 bg-gray-300 animate-pulse rounded-md my-1' />
                    </div>
                </div>
            </div>
        </div>
    );
};
