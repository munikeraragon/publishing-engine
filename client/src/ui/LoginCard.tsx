export interface LoginCardProps {
    className?: string;
    buttons: LoginCardButton[];
    privacyHref: string;
    termsHref: string;
}

export interface LoginCardButton {
    href: string;
    icon: JSX.Element;
    label: string;
}

export const LoginCard: React.FC<LoginCardProps> = ({
    className = '',
    buttons,
    privacyHref,
    termsHref
}) => {
    return (
        <div
            data-aos='fade-up'
            className={`${className} flex mx-4 my-20 md:m-auto flex-col bg-white p-10 
                        gap-6 bg-primary-800 sm:rounded-md z-10 sm:max-w-sm shadow-lg`}>
            <div className='flex gap-2 flex-col'>
                <span className='text-3xl font-bold'>Welcome</span>
                <div className='flex-wrap'>
                    By logging in you accept our
                    <a href={privacyHref} className='text-indigo-600 hover:underline'>
                        {' '}
                        Privacy Policy{' '}
                    </a>
                    and
                    <a href={termsHref} className='text-indigo-600 hover:underline'>
                        {' '}
                        Terms of Service
                    </a>
                    .
                </div>
            </div>

            <div className='flex flex-col gap-5'>
                {buttons.map((b: LoginCardButton, index: number) => (
                    <a href={b.href} key={index}>
                        <div className='flex justify-center py-3 bg-indigo-500 hover:bg-gray-700 text-white rounded-md mt-2'>
                            {b.icon}
                            <span className='ml-2 font-bold'>{b.label}</span>
                        </div>
                    </a>
                ))}
            </div>
        </div>
    );
};
