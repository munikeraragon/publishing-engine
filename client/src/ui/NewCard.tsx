import { Tags } from './article/Tags';

export interface Props {
    src: string | undefined | null;
    title: string | undefined;
    description: string | undefined;
    userName: string | undefined;
    userIcon: string | undefined;
    creationDate: string;
    readingTime: number;
    tags: string[];
    className?: string;
}

export const NewCard: React.FC<Props> = ({
    src,
    title,
    description,
    userIcon,
    userName,
    creationDate,
    readingTime,
    tags,
    className = ''
}) => {
    return (
        <div
            className='flex flex-col overflow-hidden rounded-lg shadow-lg
            transition transform hover:-translate-y-1.5 hover:shadow-2xl'
            style={{ height: 480 }}>
            <div className='flex-shrink-0'>
                <a href=''>
                    <div className='relative' style={{ paddingBottom: '56.25%' }}>
                        <div className='absolute inset-0'>
                            <img className=' w-full h-full object-cover' src={src} />
                        </div>
                    </div>
                </a>
            </div>

            <div className='flex flex-col justify-between flex-1 p-6 bg-white'>
                <div className='flex flex-col flex-1'>
                    <a className='block hover:underline'>
                        <h3 className='mt-2 text-lg font-semibold leading-7 text-gray-900 md:text-xl'>
                            {title}
                        </h3>
                    </a>
                    <p className='my-3 text-base leading-5 text-gray-700 sm:text-base'>
                        {description}
                    </p>

                    <div className='mt-auto'>
                        <Tags tags={tags} size='sm' />
                    </div>
                </div>
                <div className='flex items-center mt-6'>
                    <div className='flex-shrink-0'>
                        <img src={userIcon} className='inline-block h-10 w-10 rounded-full' />
                    </div>
                    <div className='flex flex-col ml-3 text-sm font-medium leading-5'>
                        <p className='text-gray-700'>{userName}</p>
                        <p className='text-gray-500'>
                            {creationDate}
                            <span className='hidden mx-2 sm:inline'>Â·</span>
                            <span className='block sm:inline'>{readingTime} min read</span>
                        </p>
                    </div>
                </div>
            </div>
        </div>
    );
};
