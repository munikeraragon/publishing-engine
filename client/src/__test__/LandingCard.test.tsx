import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { LandingCard, LandingCardProps } from '../ui/LandingCard';

const renderLandingCard = (props: Partial<LandingCardProps> = {}) => {
    const defaultProps: LandingCardProps = {
        primaryHeader: 'Welcome!',
        secondaryHeader: "Let's Start Coding",
        description: 'This is a description',
        linkLabel1: 'See Projects',
        linkHref1: '#projects',
        linkLabel2: 'Read Articles',
        linkHref2: '/articles'
    };
    return render(<LandingCard {...defaultProps} {...props} />);
};

describe('<LandingCard />', () => {
    afterEach(cleanup);

    test('renders without crashing', () => {
        renderLandingCard();
    });

    test('renders the correct content', () => {
        const { getByText } = renderLandingCard();
        getByText('Welcome!');
        getByText("Let's Start Coding");
        getByText('This is a description');
    });

    test('matches Progress card snapshot', () => {
        const tree = renderer
            .create(
                <LandingCard
                    primaryHeader='Welcome!'
                    secondaryHeader="Let's Start Coding"
                    description='This is a description'
                    linkLabel1='See Projects'
                    linkHref1='#projects'
                    linkLabel2='Read Articles'
                    linkHref2='/articles'
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
