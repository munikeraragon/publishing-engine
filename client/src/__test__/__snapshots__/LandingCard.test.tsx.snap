// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingCard /> matches Progress card snapshot 1`] = `
<div
  className="flex flex-col justify-center mx-4 "
>
  <div
    className="sm:text-center lg:text-left"
  >
    <h1
      className="text-4xl tracking-tight font-bold text-gray-700"
    >
      Welcome!
    </h1>
    <h1
      className="text-4xl font-bold text-indigo-600"
    >
      Let's Start Coding
    </h1>
    <p
      className="mt-3 text-base text-gray-500 sm:mt-5 sm:text-lg sm:max-w-xl sm:mx-auto md:mt-5 md:text-xl lg:mx-0"
    >
      This is a description
    </p>
    <div
      className="mt-5 sm:mt-8 sm:flex sm:justify-center lg:justify-start"
    >
      <div
        className="rounded-md shadow"
      >
        <span
          className="w-full flex items-center justify-center px-8 py-2 border border-transparent text-base font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-700  md:py-2 md:text-lg md:px-10"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          See Projects
        </span>
      </div>
      <div
        className="mt-3 sm:mt-0 sm:ml-3"
      >
        <span
          className="w-full flex items-center justify-center px-8 py-3 border border-transparent text-base font-medium rounded-md text-indigo-700 bg-indigo-100 hover:bg-indigo-200 hover:bg-blue-200 md:py-2 md:text-lg md:px-10"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Read Articles
        </span>
      </div>
    </div>
  </div>
</div>
`;
